apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx

```diff
@@ -0,0 +1,183 @@
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { isConfirmedTx } from '@alephium/shared'
+import { useInterval } from '@alephium/shared-react'
+import { useQuery } from '@tanstack/react-query'
+import { t } from 'i18next'
+import { X } from 'lucide-react'
+import { memo, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { fadeInBottom, fadeOut } from '@/animations'
+import { pendingTransactionQuery } from '@/api/queries/transactionQueries'
+import Button from '@/components/Button'
+import HashEllipsed from '@/components/HashEllipsed'
+import { openModal } from '@/features/modals/modalActions'
+import { selectTopModal } from '@/features/modals/modalSelectors'
+import { sentTransactionStatusChanged } from '@/features/sentTransactions/sentTransactionsActions'
+import { selectSentTransactionByHash } from '@/features/sentTransactions/sentTransactionsSelectors'
+import SnackbarBox from '@/features/snackbar/SnackbarBox'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import { SentTransaction } from '@/types/transactions'
+
+const SentTransactionSnackbarPopup = memo(({ txHash }: { txHash: string }) => {
+  const dispatch = useAppDispatch()
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+  const [hide, setHide] = useState(false)
+
+  const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
+
+  useEffect(() => {
+    if (!tx) return
+
+    dispatch(sentTransactionStatusChanged({ hash: tx.hash, status: isConfirmedTx(tx) ? 'confirmed' : 'mempooled' }))
+  }, [dispatch, tx])
+
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>
+
+    if (sentTx?.status === 'confirmed') {
+      timer = setTimeout(() => setHide(true), 5000)
+    }
+
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, [sentTx?.status])
+
+  if (!sentTx || hide) return null
+
+  return (
+    <SentTransactionSnackbarPopupStyled {...fadeInBottom} {...fadeOut} className="info">
+      <Columns>
+        <Rows>
+          <Message status={sentTx.status} />
+
+          <HashEllipsedStyled hash={txHash} tooltipText={t('Copy hash')} showSnackbarOnCopied={false} />
+
+          <Progress status={sentTx.status} />
+        </Rows>
+        <ButtonsRow>
+          {sentTx.status !== 'sent' && <DetailsButton hash={txHash} />}
```

Comment:
Note: I'm going to redesign this a bit, but the idea to open the TX modal is great!